vllm/sampling_params.py

```diff
@@ -28,6 +28,77 @@ class SamplingType(IntEnum):
 to sample from."""
 
 
+class NoBadWordsLogitsProcessor:
+    _SMALLEST_LOGIT = float("-inf")
+    _NEUTRAL_LOGIT = 0.0
+
+    def __init__(self, bad_words_ids: List[List[int]]):
+        self.bad_words_ids = bad_words_ids
+        self.word_bias: torch.FloatTensor = None
+
+    def __call__(
+        self,
+        past_tokens_ids: List[int],
+        logits: torch.FloatTensor,
+    ) -> torch.Tensor:
+        if self.word_bias is None:
+            self._init_word_bias(logits=logits)
+
+        last_token_bias = torch.zeros_like(logits)
+
+        for bad_word_ids in self.bad_words_ids:
+            if len(bad_word_ids) == 1:  # 1-token words already processed
+                continue
+
+            if len(bad_word_ids) > len(past_tokens_ids) + 1:
```

Comment:
This differs from the original inequality here: https://github.com/huggingface/transformers/blob/main/src/transformers/generation/logits_process.py#L1131
```python
if len(sequence_ids) > input_ids.shape[1]:  # the sequence is longer than the context, ignore
    continue
```

I may be wrong, but it seemed we should allow for one token to be generated (so + 1).